x-pack/libbeat/management/tests/mbtest/system/process_integration_test.go

```diff
@@ -0,0 +1,220 @@
+		Id:             unitOutID,
+		Type:           proto.UnitType_OUTPUT,
+		ConfigStateIdx: 1,
+		State:          proto.State_HEALTHY,
+		Config: &proto.UnitExpectedConfig{
+			DataStream: &proto.DataStream{
+				Namespace: "default",
+			},
+			Type:     "file",
+			Revision: 1,
+			Meta: &proto.Meta{
+				Package: &proto.Package{
+					Name:    "system",
+					Version: "1.17.0",
+				},
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
+				"type":            "file",
+				"enabled":         true,
+				"path":            outPath,
+				"filename":        "beat-out",
+				"number_of_files": 7,
+			}),
+		},
+	}
+
+	observedStates := make(chan *proto.CheckinObserved)
+	expectedUnits := make(chan []*proto.UnitExpected)
+	done := make(chan struct{})
+	// V2 mock server
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			select {
+			case observedStates <- observed:
+				return &proto.CheckinExpected{
+					Units: <-expectedUnits,
+				}
+			case <-done:
+				return nil
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+	}
+	require.NoError(t, server.Start())
+	defer server.Stop()
+
+	// start the client
+	client := client.NewV2(fmt.Sprintf(":%d", server.Port), token, client.VersionInfo{
+		Name: "program",
+		Meta: map[string]string{
+			"key": "value",
+		},
+	}, client.WithGRPCDialOptions(grpc.WithTransportCredentials(insecure.NewCredentials())))
+
+	lbmanagement.SetManagerFactory(func(cfg *conf.C, registry *reload.Registry) (lbmanagement.Manager, error) {
+		c := management.DefaultConfig()
+		if err := cfg.Unpack(&c); err != nil {
+			return nil, err
+		}
+		return management.NewV2AgentManagerWithClient(c, registry, client, management.WithStopOnEmptyUnits)
+	})
+
+	go func() {
+		t.Logf("Running beats...")
+		err := cmd.RootCmd.Execute()
+		require.NoError(t, err)
+	}()
+
+	scenarios := []struct {
+		expectedStatus proto.State
+		nextInputunit  *proto.UnitExpected
+	}{
+		{
+			proto.State_HEALTHY,
+			&inputStreamIncorrectPid,
+		},
+		{
+			proto.State_DEGRADED,
+			&inputStreamCorrectPid,
+		},
+		{
+			proto.State_HEALTHY,
+			&inputStreamCorrectPid,
+		},
+		// wait for one more checkin, just to be sure it's healthy
+		{
+			proto.State_HEALTHY,
+			&inputStreamCorrectPid,
+		},
+	}
+
+	timer := time.NewTimer(2 * time.Minute)
+	id := 0
+	for id < len(scenarios) {
+		select {
+		case observed := <-observedStates:
+			state := extractState(observed.GetUnits(), unitOneID)
+			expectedUnits <- []*proto.UnitExpected{
+				scenarios[id].nextInputunit,
+				&outputExpectedStream,
+			}
+			if state != scenarios[id].expectedStatus {
+				continue
+			}
+			// always ensure that output is healthy
+			outputState := extractState(observed.GetUnits(), unitOutID)
+			require.Equal(t, outputState, proto.State_HEALTHY)
+
+			timer.Reset(2 * time.Minute)
+			id++
+		case <-timer.C:
+			t.Fatal("timeout waiting for checkin")
```

Comment:
[Suggestion]
Add the timeout in the error message
```suggestion
			t.Fatalf("timeout after %s waiting for checkin", timeout)
```

[Consistency]
our tests basically all uses testify, I'd recommend to use it, mainly in tests like this one, where you're already using it, so stick with it instead of mixing with the std lib.

Reply:
Done.